modules/statistics/RTPStatsCollector.js

```diff
@@ -309,6 +310,38 @@ StatsCollector.prototype._processAndEmitReport = function() {
                 videoCodec = ssrcStats.codec;
             }
 
+            // Mix { sourceName, { ssrc, resolution } } to resolutions,
+            // which becomes { participantId | sourceName, { ssrc, resolution } }.
+            // Mix { sourceName, { ssrc, framerate } } to framerates,
+            // which becomes participantId | sourceName, { ssrc, resolution } }.
+            // This enables jitsi-meet code to select resolution and framerate by
+            // either participantId or sourceName.
+            if (FeatureFlags.isSourceNameSignalingEnabled()) {
+                const sourceName = track.getSourceName();
+
+                if (sourceName) {
+                    const resolution = ssrcStats.resolution;
+
+                    if (resolution.width // eslint-disable-line max-depth
+                        && resolution.height
+                        && resolution.width !== -1
+                        && resolution.height !== -1) {
+                        const sourceResolutions = resolutions[sourceName] || {};
+
+                        sourceResolutions[ssrc] = resolution;
```

Comment:
Can `sourceResolutions` be a map <sourceName, resolution> instead ? Same with `sourceFramerates` as well. SSRCs for a given track can change because of an upcoming feature. This can also be made in a follow-up PR since this current implementation is good for the release.

Reply:
They can be maps <sourceName, resolution> but jitsi-meet also need to change to adapt to the new structure. e.g. https://github.com/jitsi/jitsi-meet/blob/master/react/features/connection-stats/components/ConnectionStatsTable.js#L656
I think a separate PR for this structural change is a good idea.